go/cmd/vtctldclient/command/vreplication/common/utils.go

```diff
@@ -260,29 +261,50 @@ func GetTabletSelectionPreference(cmd *cobra.Command) tabletmanagerdatapb.Tablet
 }
 
 func OutputStatusResponse(resp *vtctldatapb.WorkflowStatusResponse, format string) error {
-	var output []byte
-	var err error
 	if format == "json" {
-		output, err = cli.MarshalJSONPretty(resp)
+		output, err := cli.MarshalJSONPretty(resp)
 		if err != nil {
 			return err
 		}
-	} else {
-		tout := bytes.Buffer{}
-		tout.WriteString(fmt.Sprintf("The following vreplication streams exist for workflow %s.%s:\n\n",
-			BaseOptions.TargetKeyspace, BaseOptions.Workflow))
-		for _, shardstreams := range resp.ShardStreams {
-			for _, shardstream := range shardstreams.Streams {
-				tablet := fmt.Sprintf("%s-%d", shardstream.Tablet.Cell, shardstream.Tablet.Uid)
-				tout.WriteString(fmt.Sprintf("id=%d on %s/%s: Status: %s. %s.\n",
-					shardstream.Id, BaseOptions.TargetKeyspace, tablet, shardstream.Status, shardstream.Info))
-			}
+		fmt.Println(string(output))
+		return nil
+	}
+
+	// Plain text formatted output.
+	tout := strings.Builder{}
+	tout.WriteString(fmt.Sprintf("The following vreplication streams exist for workflow %s.%s:\n\n",
+		BaseOptions.TargetKeyspace, BaseOptions.Workflow))
+	for _, shardstreams := range resp.ShardStreams {
+		for _, shardstream := range shardstreams.Streams {
+			tablet := fmt.Sprintf("%s-%d", shardstream.Tablet.Cell, shardstream.Tablet.Uid)
+			tout.WriteString(fmt.Sprintf("id=%d on %s/%s: Status: %s. %s.\n",
+				shardstream.Id, BaseOptions.TargetKeyspace, tablet, shardstream.Status, shardstream.Info))
+		}
+	}
+	if len(resp.TableCopyState) > 0 {
```

Comment:
This check here makes the change backwards / forwards compatible, right?

Reply:
Well, I suppose that's true. But previously we only returned/printed the table copy state in the json formatted output (`--format=json`). Now we're including it, if there is any, in the plain text formatted output as well.

Reply:
No, but this commit does (the `vtctlservice` RPC used here has always had the `TableCopyState` field) 🙂  https://github.com/vitessio/vitess/pull/18825/commits/adab0037ca34f2d1f2b3f9ea8ce507780886d0d3

Thanks! ❤️